documentation/api/addType.md

```diff
@@ -57,6 +59,13 @@ instances. The name of the type should be `Person` and it should
 inherit from the built-in `object` type. Furthermore we add an
 `identify` method that will recognize `Person` instances.
 
+<!-- unexpected-markdown hide:true -->
```

Comment:
This is great!

Reply:
Could it be `freshContext: true` instead, though, maybe with some extra work? Just so we try to keep presentation and evaluation separate. (If we want to switch to another mechanism for generating the site, that would have to support `<!-- unexpected-markdown hide:true -->`)

Reply:
This particular case has special constraints, specifically that at some point in the file we need to change to a newer version of the type and then use it for the rest of the file. That’s why this one looks the way it does.

Reply:
I thought that was how essentially what `freshContext` was supposed to do? Like if you have 3 code snippets in a file:

````
```js
// A
expect.addType(theOldType);
```

<!-- evaldown freshContext:true -->
```js
// B
expect.addType(theNewType);
```

```js
// C
// Here the new type is still in effect
```
````

The way I imagine it, `freshContext` creates a boundary so that A doesn't bleed into B, but C can still see everything that B introduced?

Reply:
Hmmm, so you wanted freshContext to be created for that particular snippet but then last beyond it? I guess I could make that work, but it introduces an inconsistency with the way the rest of the flags with the "scope" of the other flags (with the exception of nowrap). If it were to behave that way I think I'd prefer requiring nowrap also be specified.

Reply:
Yeah, I think having it work like that would make sense. Fairly sure that's also how `freshExpect` works. So it essentially creates a "boundary" where all previous blocks don't bleed in.

What is `nowrap`?

Reply:
nowrap means expose anything created in the block to subsequent blocks - you need a flag like that if you default to blocks being isolated to allow anything you define to survive :)

Reply:
I don't understand how that's different from the next block having `freshContext:true`? Is it necessary to have two flags?

Reply:
I'm not sure where the confusion is to let me just explain how things are set up now.

By default, all snippets are wrapped in functions. That means no locals can escape. In order to allow locals to escape into the next snippet, I added an explicit `nowrap` flag - which basically requests executing in such a way the definitions persist.

When`freshContext` was proposed I made it rebuild the global state from scratch, but note:
- _currently_ it creates the new context only for a single snippet
- the snippet is still wrapped in a function so anything defined within that scope will still be thrown away

I'm open to considering allowing freshContext to alter all following snippets - despite the fact I don't like that you can't get back to the original global context. But, you're still going to need to use the `nowrap` flag to have anything defined by the execution survive.

Reply:
Definitions now persist until the end of the file which has removed the need for the `nowrap` flag.